### DISCUSSION TITLE
How can I fixed xcodebuild fail...

### DISCUSSION THREAD

**Discussion by jh0152park** [appium/appium#18249] [Q&A]
my condition
Xcode 14.2
appium 1.22.3
idk 17.0.6
Mac m1 Mac Pro 13.0.1

I spend entire my day to try fixed below error however its hard to me, could you give me more detail guide please?
thank you and best regards 

1. occurred cannot link directly error as below

when try to fixed /Applications/Appium\ Server\ GUI.app/Contents/Resources/app/node_modules/appium/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj file (signing team, bundle idnetifier) occurred error as below when press run button

ld: cannot link directly with dylib/framework, your binary is not an allowed client of /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework/XCTAutomationSupport for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

2. occurred xcodebuild failed with code 65 when I push the Start Session button at Appium Inspector as below

[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[WD Proxy] connect ECONNREFUSED 127.0.0.1:8100
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[WD Proxy] connect ECONNREFUSED 127.0.0.1:8100
[WebDriverAgent] xcodebuild exited with code '65' and signal 'null'
[BaseDriver] Event 'wdaStartFailed' logged at 1677248292069 (23:18:12 GMT+0900 (Korean Standard Time))
[XCUITest] Unable to launch WebDriverAgent because of xcodebuild failure: xcodebuild failed with code 65
[XCUITest] xcodebuild error message:
[XCUITest] 
[XCUITest] Quitting and uninstalling WebDriverAgent
[WebDriverAgent] Shutting down sub-processes
[iOSSim] Building bundle path map
[iOSSim] The simulator has '0' bundles which have 'WebDriverAgentRunner-Runner' as their 'CFBundleName':
[WebDriverAgent] No WDAs on the device.
[XCUITest] {}
[DevCon Factory] Releasing connections for 978FB1B8-8119-4EFD-8582-D9BF92D71025 device on any port number
[DevCon Factory] Found cached connections to release: ["978FB1B8-8119-4EFD-8582-D9BF92D71025:8100"]
[DevCon Factory] Cached connections count: 0
[XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[IOSSimulatorLog] Stopping iOS log capture
[BaseDriver] Event 'newSessionStarted' logged at 1677248292087 (23:18:12 GMT+0900 (Korean Standard Time))
[W3C] Encountered internal error running command: Error: Unable to launch WebDriverAgent because of xcodebuild failure: xcodebuild failed with code 65
[W3C] xcodebuild error message:
[W3C] 
[W3C]     at quitAndUninstall (/Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/driver.js:585:15)
[W3C]     at /Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/driver.js:620:11
[W3C]     at wrapped (/Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/asyncbox/lib/asyncbox.js:60:13)
[W3C]     at retry (/Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/asyncbox/lib/asyncbox.js:43:13)
[W3C]     at retryInterval (/Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/asyncbox/lib/asyncbox.js:70:10)
[W3C]     at /Applications/Appium Server GUI.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/driver.js:600:7

**Comment by jlipps** [comment]
Appium 1 is not supported. Use Appium 2 and the latest xcuitest driver.

**Reply by jh0152park** [reply]
Thank you for your reply
Currently I tried re install Appium 1 to 2 as below and same error still occurred.
Please help me with more detail guide please... T.T also I keep trying fixed it

```
$ appium -v
2.0.0-beta.56

$ appium driver list --installed
- xcuitest@4.18.2 [installed (NPM)]

$ appium-doctor --version
1.16.2

WARN AppiumDoctor [Deprecated] Please use appium-doctor installed with "npm install @appium/doctor --location=global"
info AppiumDoctor Appium Doctor v.1.16.2
info AppiumDoctor ### Diagnostic for necessary dependencies starting ###
info AppiumDoctor  ✔ The Node.js binary was found at: /opt/homebrew/bin/node
info AppiumDoctor  ✔ Node version is 19.7.0
info AppiumDoctor  ✔ Xcode is installed at: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ Xcode Command Line Tools are installed in: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ DevToolsSecurity is enabled.
info AppiumDoctor  ✔ The Authorization DB is set up properly.
info AppiumDoctor  ✔ Carthage was found at: /opt/homebrew/bin/carthage. Installed version is: 0.39.0
info AppiumDoctor  ✔ HOME is set to: /Users/parkjaehyeon
info AppiumDoctor  ✔ ANDROID_HOME is set to: /Users/parkjaehyeon/Library/Android/sdk
info AppiumDoctor  ✔ JAVA_HOME is set to: /Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home
info AppiumDoctor    Checking adb, android, emulator, apkanalyzer
info AppiumDoctor      'adb' is in /Users/parkjaehyeon/Library/Android/sdk/platform-tools/adb
info AppiumDoctor      'android' is in /Users/parkjaehyeon/Library/Android/sdk/tools/android
info AppiumDoctor      'emulator' is in /Users/parkjaehyeon/Library/Android/sdk/emulator/emulator
info AppiumDoctor      'apkanalyzer' is in /Users/parkjaehyeon/Library/Android/sdk/cmdline-tools/latest/bin/apkanalyzer
info AppiumDoctor  ✔ adb, android, emulator, apkanalyzer exist: /Users/parkjaehyeon/Library/Android/sdk
info AppiumDoctor  ✔ 'bin' subfolder exists under '/Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home'
info AppiumDoctor ### Diagnostic for necessary dependencies completed, no fix needed. ###
```

also Appium desktop and inspector already used latest version 

thanks

**Reply by KazuCocoa** [reply]
Could you share the full xcodebuild log, or `"appium:showXcodeLog": true` full appium log as GIST?


Btw, please use newer appium doctor as below deprecation message: (The result itself shouldn't different so much
> WARN AppiumDoctor [Deprecated] Please use appium-doctor installed with "npm install @appium/doctor --location=global"
